paasta_tools/kubernetes/application/controller_wrappers.py

```diff
@@ -196,12 +199,35 @@ def create(self, kube_client: KubeClient) -> None:
         self.ensure_pod_disruption_budget(kube_client)
         self.sync_horizontal_pod_autoscaler(kube_client)
 
+    def deep_delete_and_create(self, kube_client: KubeClient) -> None:
+        self.logging.debug(
+            f"Deep Delete {self.kube_deployment.service} {self.kube_deployment.instance}."
+        )
+        self.deep_delete(kube_client)
+        while self.kube_deployment in set(list_all_deployments(kube_client)):
+            self.logging.debug(
+                f"waiting for instance {self.kube_deployment.service} {self.kube_deployment.instance} to die."
+            )
+            sleep(0.5)
+        self.logging.debug(
+            f"{self.kube_deployment.service} {self.kube_deployment.instance} are deleted, creating new one."
+        )
+        create_deployment(kube_client=kube_client, formatted_deployment=self.item)
```

Comment:
I think you can directly call self.create here? And return after the thread starts.